packages/node/src/integrations/tracing/nest/sentry-nest-instrumentation.ts

```diff
@@ -163,33 +163,84 @@ export class SentryNestInstrumentation extends InstrumentationBase {
+                  // eslint-disable-next-line @typescript-eslint/unbound-method
+                  next.handle = new Proxy(next.handle, {
+                    apply: (originalHandle, thisArgHandle, argsHandle) => {
+                      beforeSpan.end();
+
+                      if (parentSpan) {
+                        return withActiveSpan(parentSpan, () => {
+                          const handleReturnObservable = Reflect.apply(originalHandle, thisArgHandle, argsHandle);
+
+                          if (!request._sentryInterceptorInstrumented) {
+                            afterSpan = startInactiveSpan(
+                              getMiddlewareSpanOptions(target, 'Interceptor - After Route'),
+                            );
                           }
-                        };
-                      }
 
-                      return Reflect.get(target, property, receiver);
+                          return handleReturnObservable;
+                        });
+                      } else {
+                        const handleReturnObservable = Reflect.apply(originalHandle, thisArgHandle, argsHandle);
+
+                        if (!request._sentryInterceptorInstrumented) {
+                          afterSpan = startInactiveSpan(getMiddlewareSpanOptions(target, 'Interceptor - After Route'));
+                        }
+
+                        return handleReturnObservable;
+                      }
                     },
                   });
 
-                  return originalIntercept.apply(thisArgIntercept, [executionContext, nextProxy, args]);
+                  let returnedObservableInterceptMaybePromise: Observable<unknown> | Promise<Observable<unknown>>;
+
+                  try {
+                    returnedObservableInterceptMaybePromise = originalIntercept.apply(thisArgIntercept, argsIntercept);
+                  } catch (e) {
+                    if (!request._sentryInterceptorInstrumented) {
+                      if (beforeSpan) beforeSpan.end();
+                      if (afterSpan) afterSpan.end();
+                      addNonEnumerableProperty(request, '_sentryInterceptorInstrumented', true);
+                    }
+
+                    throw e;
+                  }
+
+                  if (request._sentryInterceptorInstrumented) {
+                    return returnedObservableInterceptMaybePromise;
+                  }
+
+                  // handle async interceptor
+                  if (isThenable(returnedObservableInterceptMaybePromise)) {
+                    return Promise.resolve(returnedObservableInterceptMaybePromise).then(
+                      observable => {
+                        instrumentObservable(observable, afterSpan ?? parentSpan);
+                        return observable;
+                      },
+                      e => {
+                        if (beforeSpan) beforeSpan.end();
+                        if (afterSpan) afterSpan.end();
```

Comment:
m: Let's add brackets to the `if` statements. We don't have lint rules for it but we basically have an unwritten agreement that we use brackets for if statements. You could also do `afterSpan?.end()`.